Sources/SnapshotTestingStitch/SnapshottingHEIC.swift

```diff
@@ -0,0 +1,133 @@
+            style: style,
+            precision: precision,
+            compressionQuality: compressionQuality
+        )
+    }
+
+    /// Stitches multiple visual snapshot strategies into a single image asset.
+    ///
+    /// - Parameters:
+    ///   - tasks: The named tasks which should be carried out, in the order that they should be displayed.
+    ///   Titles will be displayed above their respectful image, allowing for easier identification.
+    ///   Any strategy can be used as long as the output format is UIImage.
+    ///   - style: The style configuration which allows for you to customise the appearance of the output image,
+    ///   including but not limited to the item spacing, and optional image borders.
+    ///   - precision: The percentage of pixels that must match in the final comparison
+    ///   in order for the test to successfully pass.
+    ///   - compressionQuality: The desired compression quality to use when writing to an image destination.
+    static func stitchHEIC(
+        strategies tasks: [(name: String, strategy: Snapshotting<Value, Format>)],
+        style: StitchStyle = .init(),
+        precision: Float = 1,
+        compressionQuality: CompressionQuality = .lossless
+    ) -> Snapshotting {
+        stitchHEIC(
+            strategies: tasks.map { .init(name: $0.name, strategy: $0.strategy, configure: nil) },
+            style: style,
+            precision: precision,
+            compressionQuality: compressionQuality
+        )
+    }
+
+    /// Stitches multiple visual snapshot strategies into a single image asset.
+    ///
+    /// - Parameters:
+    ///   - tasks: The tasks which should be carried out, in the order that they should be displayed.
+    ///   Tasks can include a title which will be displayed above their respectful image,
+    ///   allowing for easier identification. Any strategy can be used as long as the output format is UIImage.
+    ///   Tasks can can also contain a configuration block which allows for you to modify
+    ///   the value just before it's snapshot is taken.
+    ///   - style: The style configuration which allows for you to customise the appearance of the output image,
+    ///   including but not limited to the item spacing, and optional image borders.
+    ///   - precision: The percentage of pixels that must match in the final comparison
+    ///   in order for the test to successfully pass.
+    ///   - compressionQuality: The desired compression quality to use when writing to an image destination.
+    static func stitchHEIC(
+        strategies tasks: [StitchTask<Value, Format>],
+        style: StitchStyle = .init(),
+        precision: Float = 1,
+        compressionQuality: CompressionQuality = .lossless
+    ) -> Snapshotting {
+        let internalStrategy: Snapshotting<UIViewController, UIImage> = .imageHEIC(
+            precision: precision,
+            compressionQuality: compressionQuality
+        )
```

Comment:
I'm a little wary that we have so much duplication in this class for what is essentially just these four lines.

Could I maybe propose a slight change to the public API and thus would lead to a reduction in duplication?

```swift
.stitch(
  strategies: [
    ("iPhone 8", .image(on: .iPhone8)),
    ("iPhone 8 Plus", .image(on: .iPhone8Plus)),
  ],
  format: .heic(compression: .lossless) // new
)

```

We simply add (to the existing functions) a new format parameter. This would be a new enum which has the values `png` and `heic`. It would default to PNG for backwards compatibility.

Inside the stitch function we would switch on that format value in order to choose the "internal strategy".

Reply:
I agree, it was a very quick decision. Updated PR)